src/lib/filter/adjudicators/adjudicators.ts

```diff
@@ -153,8 +153,15 @@ export const definedCategory = (binding: Partial<Binding>) => {
     binding.isValidate ? "Validate" :
     "";
 };
-
-export const definedCallback = (binding: Partial<Binding>) => {
+export type DefinedCallbackReturnType =
+  | FinalizeAction<GenericClass, InstanceType<GenericClass>>
+  | WatchLogAction<GenericClass, InstanceType<GenericClass>>
+  | MutateAction<GenericClass, InstanceType<GenericClass>>
+  | ValidateAction<GenericClass, InstanceType<GenericClass>>
+  | null
+  | undefined;
```

Comment:
This is an interesting type to me, I think we can use some trend analysis across the codebase to see where we do odd things like this. But, that's for a follow up issue! Looks good here.